### ISSUE TITLE
Possible improvements for Touch Gamepad/Keyboard

### ISSUE THREAD

**Issue by ZxAsSeMbLeR** [bog-dan-ro/spectacol#105]
After trying and comparing this emu with other similars in Android for ZX Spectrum, it has a lot of good features. Anyway in an emulator in particular: Spectaculator (for Android), there is a way of managing the touch controls I really like. Maybe some ideas could also be used in Spectacol.

The virtual gamepad is on screen but after some seconds without touching the screen the gamepad hides automatically and it appears again when the screen is touched. The feeling is really nice.

When you touch the screen outside the virtual gamepad in a place with neither buttons nor options, the keyboard appears, and pressing ouside the keyboard makes the keyboard dissapear again. It is really fast to access/hide the virtual keyboard. In spectacol you have to press the corresponding little button each time you want to show/hide the keyboard. You have also to press the gamepad button to show/hide the virtual gamepad. It could be solved automatically as explained before.

Just some suggestions. Opinions? ;-)

Thanks for your great emu!

**Comment by albanlusitanae** [comment]
So you are discussing doing this in Virtual Gamepad Landscape mode right?

**Comment by ZxAsSeMbLeR** [comment]
Yes. This idea is for virtual gamepad/keyboard in landscape mode (in portrait the controls can be seen all the time)

**Comment by bog-dan-ro** [comment]
In landscape mode I think it will be very annoying, because you ca easily touch outside gamepad area. When I play a game I don't want to see the keyboard pop up on the screen.
Maybe this will work on portrait mode ...

**Comment by ZxAsSeMbLeR** [comment]
Well maybe not in every place outside the gamepad area, but in a certain part of the screen (an area bigger than just a little button on the left). For example, as the gamepad/buttons are in the botton part of the screen, maybe touching in the upper part of the screen (starting from the middle) and from the center to the left side (for not interfering with the buttons) to bring the keayboard, I think would be nicer and useful. 

There are some games with more keys to use than just the gamepad and 4 assignable buttons, or you just don't want to assign all the buttons in that moment and just press the virtual keyboard keys to make some actions like picking an object (for example 1 or 2 in Three Weeks in Paradise or Z in Atic Atac) and accessing the keyboard fast is a must.

Please try the free version of Spectaculator on Android to see in realtime what I am talking about. If after that you still don't like it...well I am not going to insist on it more ;-), but IT IS really a very nice way of virtual pad to import to your emulator, at least optionally...

Thanks!

**Comment by ZxAsSeMbLeR** [comment]
I will try to clarify it a bit better why I think some improvements could be made to the way Spectacol manages the virtual controls. Let's suppose you are playing a game which uses some additional keys and are playing it using virtual gamepad/keyboard in landscape mode (which looks nicer and bigger than portrait). Suddenly you need to press a key...let's say w, for example. You need to untouch the gamepad, press the keyboard button, which sometimes doesn't respond well to the left hand (don't know why..maybe I have bigger fingers in the left hand ;-)), so in that case I press it with the right hand in a strange movement. Then I press the w, but now I have to close the keyboard again pressing the button on the left....and to go on playing the game I must also press the gamepad button to activate it again...

Meanwhile a lot of enemies are giving a good account on me ;-) (well I could press pause...and press it again...to go on). Too many press/unpress movements.

With the method I said before the process would be: press on the upper side of the screen to show the keyboard, press w, press out of the keyboard again to hide...and that is it.

Alternatively, if there was only one button to swap controls between gamepad and keyboard instead of individual buttons for the keyboard and the gamepad, I think that would also would reduce the process.  In that case the method could be:
- There could be an option in the menus to show/hide virtual controls, just in case you are playing with a gamepad or external keyboard.
- There could also be only one button to swap between virtual keyboard and virtual gamepad (which could also be activated pressing a physical back button if it exists in the device).
- If some seconds pass and you don't touch the screen the controls could hide automatically and show again as soon you touch the screen again 
- With all this...our example would be (with the virtual gamepad/keyboard enabled in the hypothetical menu option). Press the swapping button to show the keyboard, press w and press the swapping button again to go on playing.

I think also a more natural place to the swapping (or keyboard, gamepad buttons) button would be on the top-right side of the screen. Maybe it is easier to press there.

Well...too many reflections ;-)....

Thanks for reading! :-)
